pkg/pillar/types/zedroutertypes.go

```diff
@@ -887,6 +908,40 @@ func (status *NetworkInstanceStatus) IsIpAssigned(ip net.IP) bool {
 	return false
 }
 
+// CombineErrors combines all errors raised for the network instance into one error.
+func (status *NetworkInstanceStatus) CombineErrors() (combinedErr ErrorAndTime) {
+	errorSlice := []ErrorAndTime{status.ValidationErr, status.AllocationErr,
+		status.IPConflictErr, status.MTUConflictErr, status.UplinkErr,
+		status.ReconcileErr}
+	var (
+		oldestTime      time.Time
+		highestSeverity ErrorSeverity
+		errorMsgs       []string
+	)
+	for _, err := range errorSlice {
+		if err.HasError() {
+			if oldestTime.IsZero() || err.ErrorTime.Before(oldestTime) {
+				oldestTime = err.ErrorTime
+			}
+			if err.ErrorSeverity > highestSeverity {
+				highestSeverity = err.ErrorSeverity
+			}
+			errorMsgs = append(errorMsgs, err.Error)
+		}
+	}
+	combinedErr.Error = strings.Join(errorMsgs, "\n")
+	combinedErr.ErrorSeverity = highestSeverity
+	combinedErr.ErrorTime = oldestTime
+	return combinedErr
+}
+
+// EligibleForActivate checks if there are no errors that prevent NI from being activated.
+// Note that some types of errors do not block NI activation.
```

Comment:
What are those errors? 
And I assume it is intentional that status.HasError() is not checked. Is that because it will contain the union of all of the other errors?

Reply:
Right now it is just `MTUConflictErr`. When port and NI MTU are different we flag NI with error but fallback to port MTU and activate NI with that MTU value.
When we add support for NI with multiple ports, the `UplinkErr` might also become more relaxed (TBD).
Correct, `status.HasError()` is just combination of all non-nil NI errors, used to publish single error value to other microservices and to the controller. In zedrouter we now handle different error types separately and this union is just for publishing.

Reply:
Can you spell that out in the comments instead of the vague "other"?

Reply:
Done